lib/ansible/modules/cloud/docker/docker_secret.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/python
+#
+# Copyright 2016 Red Hat | Ansible
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: docker_secret
+
+short_description: Manage docker secrets.
+
+version_added: "2.4"
+
+description:
+     - Create and remove Docker secrets in a Swarm environment. Similar to `docker secret create` and `docker secret rm`.
+     - Adds to the metadata of new secrets 'ansible_key', an encrypted hash representation of the data, which is then used
+     - in future runs to test if a secret has changed.
+     - If 'ansible_key is not present, then a secret will not be updated unless the C(force) option is set.
+     - Updates to secrets are performed by removing the secret and creating it again.
+options:
+  data:
+    description:
+      - String. The value of the secret. Required when state is C(present).
+    required: false
+  labels:
+    description:
+      - "A map of key:value meta data, where the value is expected to be a string."
+      - If new meta data is provided, or existing meta data is modified, the secret will be updated by removing it and creating it again.
+    required: false
+  force:
+    description:
+      - Boolean. Use with state C(present) to force the removal and creation of an existing secret.
```

Comment:
It's a `force`. Meaning, always remove and replace, regardless of change. I need to explain it betta though.